fs/sysfs/dir.c

```diff
@@ -40,6 +42,7 @@ void sysfs_warn_dup(struct kernfs_node *parent, const char *name)
 int sysfs_create_dir_ns(struct kobject *kobj, const void *ns)
 {
 	struct kernfs_node *parent, *kn;
+	const char *mode;
```

Comment:
This should be `umode_t`. Plus I would initialize it to `S_IRWXU` and then OR the other flags below if `!sysfs_restrict`.